### PR TITLE
[WIP] ACB-10: Make the bundle compatible with Akeneo 1.4

### PR DESCRIPTION
**Pull request by damien-carcel**
| Q                    | A
| -------------------- | ---
| Bug fix?             | No
| New feature?         | No
| BC breaks?           | Yes
| Tests pass?          | Yes
| Checkstyle issues?*  | No
| PMD issues?**        | No
| Changelog updated?   | Yes
| Fixed tickets        | ACB-10
| DB schema updated?   | No
| Migration script?    | No
| Doc PR               | No

### CODE REVIEW

File: Resources/config/models.yml [hunk 1]
```diff
@@ -1,3 +1,3 @@
 parameters:
-    pimee_automatic_classification.model.add_category_action.class: PimEnterprise\Bundle\ClassificationRuleBundle\Model\ProductAddCategoryAction
-    pimee_automatic_classification.model.set_category_action.class: PimEnterprise\Bundle\ClassificationRuleBundle\Model\ProductSetCategoryAction
+    pim_ee_classification_rule.model.add_category_action.class: PimEnterprise\Bundle\ClassificationRuleBundle\Model\ProductAddCategoryAction
+    pim_ee_classification_rule.model.set_category_action.class: PimEnterprise\Bundle\ClassificationRuleBundle\Model\ProductSetCategoryAction
```
Comment (fitn) [inline Resources/config/models.yml]: pim_ee should be replace by pimee

File: Resources/config/serializers.yml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 parameters:
-    pimee_automatic_classification.denormalizer.product_rule.add_category_action.class: PimEnterprise\Bundle\ClassificationRuleBundle\Denormalizer\ProductRule\AddCategoryActionDenormalizer
-    pimee_automatic_classification.denormalizer.product_rule.set_category_action.class: PimEnterprise\Bundle\ClassificationRuleBundle\Denormalizer\ProductRule\SetCategoryActionDenormalizer
+    pim_ee_classification_rule.denormalizer.product_rule.add_category_action.class: PimEnterprise\Bundle\ClassificationRuleBundle\Denormalizer\ProductRule\AddCategoryActionDenormalizer
+    pim_ee_classification_rule.denormalizer.product_rule.set_category_action.class: PimEnterprise\Bundle\ClassificationRuleBundle\Denormalizer\ProductRule\SetCategoryActionDenormalizer
```
Comment (fitn) [inline Resources/config/serializers.yml]: pim_ee should be replace by pimee

File: Resources/config/serializers.yml [hunk 3]
```diff
@@ -21,15 +21,15 @@ services:
...
               set_value: %pimee_catalog_rule.model.set_value_action.class%
-              add_category: %pimee_automatic_classification.model.add_category_action.class%
-              set_category: %pimee_automatic_classification.model.set_category_action.class%
+              add_category: %pim_ee_classification_rule.model.add_category_action.class%
+              set_category: %pim_ee_classification_rule.model.set_category_action.class%
 
-    pimee_automatic_classification.denormalizer.product_rule.add_category_action:
-        class: %pimee_automatic_classification.denormalizer.product_rule.add_category_action.class%
+    pim_ee_classification_rule.denormalizer.product_rule.add_category_action:
+        class: %pim_ee_classification_rule.denormalizer.product_rule.add_category_action.class%
```
Comment (fitn) [inline Resources/config/serializers.yml]: pim_ee should be replace by pimee

File: Resources/config/serializers.yml [hunk 4]
```diff
@@ -21,15 +21,15 @@ services:
...
         arguments:
-            - %pimee_automatic_classification.model.add_category_action.class%
+            - %pim_ee_classification_rule.model.add_category_action.class%
 
-    pimee_automatic_classification.denormalizer.product_rule.set_category_action:
-            class: %pimee_automatic_classification.denormalizer.product_rule.set_category_action.class%
+    pim_ee_classification_rule.denormalizer.product_rule.set_category_action:
```
Comment (fitn) [inline Resources/config/serializers.yml]: pim_ee should be replace by pimee

File: spec/PimEnterprise/Bundle/ClassificationRuleBundle/Engine/ProductRuleApplier/ProductsUpdaterSpec.php [hunk 5]
```diff
@@ -22,11 +23,13 @@ class ProductsUpdaterSpec extends ObjectBehavior
 {
     function let(
         CategoryRepositoryInterface $categoryRepository,
-        ProductUpdaterInterface $productUpdater,
+        PropertySetterInterface $propertySetter,
+        PropertyCopierInterface $propertyCopier,
```
Comment (fitn) [inline spec/PimEnterprise/Bundle/ClassificationRuleBundle/Engine/ProductRuleApplier/ProductsUpdaterSpec.php]: See if we can't call setters/adders from ProductUpdaterInterface

File: Engine/ProductRuleApplier/ProductsUpdater.php [hunk 6]
```diff
@@ -32,11 +33,12 @@ class ProductsUpdater extends BaseProductsUpdater
      * {@inheritdoc}
      */
     public function __construct(
-        ProductUpdaterInterface $productUpdater,
+        PropertySetterInterface $propertySetter,
+        PropertyCopierInterface $propertyCopier,
```
Comment (fitn) [inline Engine/ProductRuleApplier/ProductsUpdater.php]: See if we can't call setters/adders from ProductUpdaterInterface

**Comment by fitn** [comment]
Just a question otherwise GTM

File: Engine/ProductRuleApplier/ProductsUpdater.php (hunk 6 above)
Comment (MarieMinasyan) [inline Engine/ProductRuleApplier/ProductsUpdater.php]: There is no ProductUpdaterInterface

**Pull request merged**